### DISCUSSION TITLE
OVH - shared hosting - Error is triggered but mail is sent symfony/mailer 5.4.21

### DISCUSSION THREAD

**Discussion by simogeo** [symfony/symfony#57443] [Q&A]
### Symfony version(s) affected

5.4.21

### Description

I'm using a shared hosting on OVH : https://www.ovhcloud.com/en/web-hosting/shared-hosting/

When sending a message with dns set to `sendmail://default` or `native:default`,I get the errors but the mail is sent !  

Error message with dns set to `sendmail://default`  : 
```
Fatal error: Uncaught Symfony\Component\Mailer\Exception\TransportException: Process failed with exit code -1: in /home/mvggxek/www/v27/lib/comp-vendor/symfony/mailer/Transport/Smtp/Stream/ProcessStream.php:60 Stack trace: #0 /home/mvggxek/www/v27/lib/comp-vendor/symfony/mailer/Transport/Smtp/SmtpTransport.php(272): Symfony\Component\Mailer\Transport\Smtp\Stream\ProcessStream->terminate() #1 /home/mvggxek/www/v27/lib/comp-vendor/symfony/mailer/Transport/Smtp/SmtpTransport.php(359): Symfony\Component\Mailer\Transport\Smtp\SmtpTransport->stop() #2 /home/mvggxek/www/v27/plugins/l21_emailSuite/index.php(46): Symfony\Component\Mailer\Transport\Smtp\SmtpTransport->__destruct() #3 /home/mvggxek/www/v27/plugins/index.php(56): include_once('/home/mvggxek/w...') #4 /home/mvggxek/www/v27/templates/admin/2021/index.php(72): include_once('/home/mvggxek/w...') #5 /home/mvggxek/www/v27/admin/index.php(74): include_once('/home/mvggxek/w...') #6 {main} thrown in /home/mvggxek/www/v27/lib/comp-vendor/symfony/mailer/Transport/Smtp/Stream/ProcessStream.php on line 60
```

Error message with `native:default` : 

```
Fatal error: Uncaught Symfony\Component\Mailer\Exception\TransportException: Process failed with exit code -1: in /home/mvggxek/www/v27/lib/comp-vendor/symfony/mailer/Transport/Smtp/Stream/ProcessStream.php:60 Stack trace: #0 /home/mvggxek/www/v27/lib/comp-vendor/symfony/mailer/Transport/SendmailTransport.php(119): Symfony\Component\Mailer\Transport\Smtp\Stream\ProcessStream->terminate() #1 /home/mvggxek/www/v27/lib/comp-vendor/symfony/mailer/Transport/AbstractTransport.php(72): Symfony\Component\Mailer\Transport\SendmailTransport->doSend(Object(Symfony\Component\Mailer\SentMessage)) #2 /home/mvggxek/www/v27/lib/comp-vendor/symfony/mailer/Transport/SendmailTransport.php(77): Symfony\Component\Mailer\Transport\AbstractTransport->send(Object(Symfony\Component\Mailer\SentMessage), Object(Symfony\Component\Mailer\DelayedEnvelope)) #3 /home/mvggxek/www/v27/lib/comp-vendor/symfony/mailer/Mailer.php(45): Symfony\Component\Mailer\Transport\SendmailTransport->send(Object(Symfony\Component\Mime\Email), NULL) #4 /home/mvggxek/www/v in /home/mvggxek/www/v27/lib/comp-vendor/symfony/mailer/Transport/Smtp/Stream/ProcessStream.php on line 60
```

Is that a bug ? It seems to be. Please tell me how could I handle that ?

I can't update to latest version (7.x) for compatibility reason with PHP 7.4+.

Thanks

### How to reproduce

1. Install symfony/mailer to OVH shared hosting https://www.ovhcloud.com/en/web-hosting/shared-hosting/
2. set DNS to `sendmail://default`
3. send a message ...

**Comment by xabbuh** [comment]
5.4.21 is quite old. Do you observe the same behaviour with 5.4.40?

**Comment by simogeo** [comment]
Thanks for the quick reply. I'll try that and let you know

**Comment by simogeo** [comment]
After update to v5.4.40, I get the following error when sending email, with DNS set to  **'sendmail://default' AND email is not sent**  :  

`Fatal error: Uncaught Error: Call to undefined method Symfony\Component\Mailer\Transport\Smtp\Stream\ProcessStream::setInteractive() in /home/mvggxek/www/v27/lib/comp-vendor/symfony/mailer/Transport/SendmailTransport.php:67 Stack trace: #0 /home/mvggxek/www/v27/lib/comp-vendor/symfony/mailer/Transport/SendmailTransportFactory.php(24): Symfony\Component\Mailer\Transport\SendmailTransport->__construct(NULL, NULL, NULL) #1 /home/mvggxek/www/v27/lib/comp-vendor/symfony/mailer/Transport.php(172): Symfony\Component\Mailer\Transport\SendmailTransportFactory->create(Object(Symfony\Component\Mailer\Transport\Dsn)) #2 /home/mvggxek/www/v27/lib/comp-vendor/symfony/mailer/Transport.php(161): Symfony\Component\Mailer\Transport->fromDsnObject(Object(Symfony\Component\Mailer\Transport\Dsn)) #3 /home/mvggxek/www/v27/lib/comp-vendor/symfony/mailer/Transport.php(113): Symfony\Component\Mailer\Transport->parseDsn('sendmail://defa...') #4 /home/mvggxek/www/v27/lib/comp-vendor/symfony/mailer/Transport.php(74): Symfony\Component\Mailer\Transpo in /home/mvggxek/www/v27/lib/comp-vendor/symfony/mailer/Transport/SendmailTransport.php on line 67`

After update to v5.4.40, I get the following error when sending email, with DNS set to  **'native://default' BUT email is sent**  :  

`Fatal error: Uncaught Symfony\Component\Mailer\Exception\TransportException: Process failed with exit code -1: in /home/mvggxek/www/v27/lib/comp-vendor/symfony/mailer/Transport/Smtp/Stream/ProcessStream.php:60 Stack trace: #0 /home/mvggxek/www/v27/lib/comp-vendor/symfony/mailer/Transport/SendmailTransport.php(120): Symfony\Component\Mailer\Transport\Smtp\Stream\ProcessStream->terminate() #1 /home/mvggxek/www/v27/lib/comp-vendor/symfony/mailer/Transport/AbstractTransport.php(72): Symfony\Component\Mailer\Transport\SendmailTransport->doSend(Object(Symfony\Component\Mailer\SentMessage)) #2 /home/mvggxek/www/v27/lib/comp-vendor/symfony/mailer/Transport/SendmailTransport.php(78): Symfony\Component\Mailer\Transport\AbstractTransport->send(Object(Symfony\Component\Mailer\SentMessage), Object(Symfony\Component\Mailer\DelayedEnvelope)) #3 /home/mvggxek/www/v27/lib/comp-vendor/symfony/mailer/Mailer.php(45): Symfony\Component\Mailer\Transport\SendmailTransport->send(Object(Symfony\Component\Mime\Email), NULL) #4 /home/mvggxek/www/v in /home/mvggxek/www/v27/lib/comp-vendor/symfony/mailer/Transport/Smtp/Stream/ProcessStream.php on line 60`

For your information _sendmail_path_ is set to	_'/usr/sbin/sendmail -t -i'_ on global PHP config.

I'm using 7.4.33, but I experienced the same issue with v8.x. When I fall back to swiftMailer, everything works well on the same web hosting.

**Comment by simogeo** [comment]
One more try after update to v5.4.40, I get the following error when sending email, with DNS set to **'sendmail://default?command=/usr/sbin/sendmail%20-i%20-t' BUT email is sent** :

`Fatal error: Uncaught Symfony\Component\Mailer\Exception\TransportException: Process failed with exit code -1: in /home/mvggxek/www/v27/lib/comp-vendor/symfony/mailer/Transport/Smtp/Stream/ProcessStream.php:60 Stack trace: #0 /home/mvggxek/www/v27/lib/comp-vendor/symfony/mailer/Transport/SendmailTransport.php(120): Symfony\Component\Mailer\Transport\Smtp\Stream\ProcessStream->terminate() #1 /home/mvggxek/www/v27/lib/comp-vendor/symfony/mailer/Transport/AbstractTransport.php(72): Symfony\Component\Mailer\Transport\SendmailTransport->doSend(Object(Symfony\Component\Mailer\SentMessage)) #2 /home/mvggxek/www/v27/lib/comp-vendor/symfony/mailer/Transport/SendmailTransport.php(78): Symfony\Component\Mailer\Transport\AbstractTransport->send(Object(Symfony\Component\Mailer\SentMessage), Object(Symfony\Component\Mailer\DelayedEnvelope)) #3 /home/mvggxek/www/v27/lib/comp-vendor/symfony/mailer/Mailer.php(45): Symfony\Component\Mailer\Transport\SendmailTransport->send(Object(Symfony\Component\Mime\Email), NULL) #4 /home/mvggxek/www/v in /home/mvggxek/www/v27/lib/comp-vendor/symfony/mailer/Transport/Smtp/Stream/ProcessStream.php on line 60`
